darts/models/torch_forecasting_model.py

```diff
@@ -593,23 +597,28 @@ def predict_from_dataset(self,
                 batch_prediction = batch_prediction[:, :n, :]
                 batch_prediction = batch_prediction.cpu().detach().numpy()
                 
-                ts_forecasts = Parallel(n_jobs=n_jobs)(delayed(self._build_forecast_series)(prediction, input_series[0])
-                                                       for prediction, input_series in zip(batch_prediction,
-                                                                                           input_series_dataset))
+                ts_forecasts = Parallel(n_jobs=n_jobs)(
```

Comment:
This change was actually already necessary before introducing future covariates. The previous version did not handle batches properly as far as I can tell. This is another sign that we need more exhaustive tests here.